src/experiment_prototype/scan_classes/scan_class_base.py

```diff
@@ -137,77 +151,74 @@ def slice_combos_sorter(list_of_combos, all_keys):
+
+        # Check all slice ids and add any missing ones to their own set.
+        for key in all_keys:
+            slice_in_set = False
+            for disjoint_set in disjoint_sets:
+                if key in disjoint_set:     # This slice interfaces with others
+                    slice_in_set = True
                     break
-            else:  # no break
-                list_of_combos.append([slice_id])
-                # Append the slice on its own, it is in its own object.
+            if not slice_in_set:    # This slice doesn't interface with the others, make a new set
+                disjoint_sets.append({key})
+
+        # Go through all sets and make sure they are disjoint (don't share any slice ids)
+        for i in range(len(disjoint_sets)):
+            for j in range(i + 1, len(disjoint_sets)):
+                bad_slices = disjoint_sets[i].intersection(disjoint_sets[j])
+                if len(bad_slices) != 0:
+                    raise ExperimentException(f"The following slices do not interface well with other slices: "
+                                              f"{bad_slices}")
+            disjoint_sets[i] = sorted(list(disjoint_sets[i]))   # Convert to a list
+
+        return disjoint_sets
+
+    def get_nested_slice_ids(self):
+        """
+        Organize the slice_ids by interface.
+
+        This method is inherited by child classes and organizes all slices in each child class which
+        should be combined by the class. For example, all slices in a Scan should be combined if they
+        share an AveragingPeriod or Sequence or are concurrent.
+
+        Returns a list of lists where each inner list contains the slices that are combined inside
+        this object. e.g. for ScanClassBase:
+        len(nested_slice_list) = # of scans in this experiment,
+        len(nested_slice_list[0]) = # of slices in the first scan
+
+        :returns:   A list that has one element per scan. Each element is a list of slice_ids
+                    signifying which slices are combined inside that scan. The list returned could
+                    be of length 1, meaning only one scan is present in the experiment.
+        :rtype:     list of lists
+        """
+        nested_combos = []
+
+        combine_below_dict = {
+            'ScanClassBase': ['AVEPERIOD', 'SEQUENCE', 'CONCURRENT'],   # Combine everything except SCAN interfaced
+            'Scan': ['SEQUENCE', 'CONCURRENT'],     # Combine everything SEQUENCE or CONCURRENT interfaced
+            'AveragingPeriod': ['CONCURRENT'],      # Combine everything CONCURRENT interfaced
+            'Sequence': []  # All slices in a Sequence are already CONCURRENT and should be combined already
+        }
+
+        combine_list = combine_below_dict[type(self).__name__]     # Returns the class name of the calling instance
 
-        list_of_combos = sorted(list_of_combos)
-        return list_of_combos
+        for k, interface_value in self.interface.items():
```

Comment:
```suggestion
        for slice_ids_combo, interface_value in self.interface.items():
```

Reply:
Could use a var name that describes what the key actually is

Reply:
Good idea